From 453919c658b90d29600f919a607aecf9dae62fd4 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Thu, 6 Sep 2018 18:26:04 +0800
Subject: [PATCH 4555/5242] MLK-19418-3: dmaengine: imx-sdma: remove context
 load in sdma_channel_resume

commit  9424bb5b8d0dd40970f6d405d49539f6bcb175cd from
https://source.codeaurora.org/external/imx/linux-imx.git

Since sdma_resume will restore all context in case sdma power off after
suspend, sdma_channel_resume does not need to restore again. Besides,
'suspend_off' can be removed too.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/imx-sdma.c |   15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index d59eb26..2e0071e 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -437,7 +437,6 @@ struct sdma_engine {
 	dma_addr_t			bd0_phys;
 	bool				bd0_iram;
 	struct sdma_buffer_descriptor	*bd0;
-	bool				suspend_off;
 	bool				fw_loaded;
 	int				idx;
 	/* clock ration for AHB:SDMA core. 1:1 is 1, 2:1 is 0*/
@@ -1403,22 +1402,11 @@ static int sdma_channel_pause(struct dma_chan *chan)
 static int sdma_channel_resume(struct dma_chan *chan)
 {
 	struct sdma_channel *sdmac = to_sdma_chan(chan);
-	struct sdma_engine *sdma = sdmac->sdma;
 	unsigned long flags;
 
 	if (!(sdmac->flags & IMX_DMA_SG_LOOP))
 		return -EINVAL;
 
-	/*
-	 * restore back context since context may loss if mega/fast OFF
-	 */
-	if (sdma->suspend_off) {
-		if (sdma_load_context(sdmac)) {
-			dev_err(sdmac->sdma->dev, "context load failed.\n");
-			return -EINVAL;
-		}
-	}
-
 	sdma_enable_channel(sdmac->sdma, sdmac->channel);
 	spin_lock_irqsave(&sdmac->vc.lock, flags);
 	sdmac->status = DMA_IN_PROGRESS;
@@ -2505,8 +2493,6 @@ static int sdma_suspend(struct device *dev)
 	struct sdma_engine *sdma = platform_get_drvdata(pdev);
 	int i, ret = 0;
 
-	sdma->suspend_off = false;
-
 	/* Do nothing if not i.MX6SX or i.MX7D*/
 	if (sdma->drvdata != &sdma_imx6sx && sdma->drvdata != &sdma_imx7d
 	    && sdma->drvdata != &sdma_imx6ul)
@@ -2563,7 +2549,6 @@ static int sdma_resume(struct device *dev)
 
 	/* Firmware was lost, mark as "not ready" */
 	sdma->fw_loaded = false;
-	sdma->suspend_off = true;
 
 	/* restore regs and load firmware */
 	for (i = 0; i < MXC_SDMA_SAVED_REG_NUM; i++) {
-- 
1.7.9.5

